chain/ethereum/src/ethereum_adapter.rs

```diff
@@ -1846,3 +1790,64 @@ async fn filter_call_triggers_from_unsuccessful_transactions(
     }
     Ok(block)
 }
+
+async fn resolve_transaction_receipt(
+    web3: Arc<Web3<Transport>>,
+    transaction_hash: H256,
+    block_hash: H256,
+    logger: Logger,
+) -> Result<TransactionReceipt, IngestorError> {
+    let retry_result = retry("batch eth_getTransactionReceipt RPC call", &logger)
+        .limit(*REQUEST_RETRIES)
+        .no_logging()
+        .timeout_secs(*JSON_RPC_TIMEOUT)
+        .run(move || web3.eth().transaction_receipt(transaction_hash))
+        .await;
+
+    match retry_result {
```

Comment:
Great comments down below 👏

Reply:
I agree, but they're not mine, though :sweat_smile: